README.md

```diff
@@ -9,11 +9,49 @@ modules that read chain state directly.
 
 ## Versioning
 
-Blockchain state versioning does not naturally align with common semantic versioning conventions.
+We adopt a policy similar to the [Builtin-Actors repository](https://github.com/filecoin-project/builtin-actors?tab=readme-ov-file#versioning), with a key difference:
 
-Any change in behaviour, including repairing any error that may have affected blockchain evaluation,
-must be released in a major version change. We intend that to be a rare event for the contents of 
-this repository.
+- The minor number in the version correlates with the `ActorVersion` from the Builtin-Actors repository.
+- We generally don't use major versions; these are always set to `0`.
+- We strive for round minor versions to denote the definitive release for a given network upgrade. However, due to the unpredictability of software engineering, further releases may be made by bumping the patch number.
+- Development versions use qualifiers like `-dev` (development) and `-rc` (release candidate).
```

Comment:
```suggestion
- The minor number in the version correlates with the `ActorVersion` from the Builtin-Actors repository.  (Builtin-Actors uses the major version number for this.)
- We don't use major versions; these are always set to `0`.  (This is because of the go's special handling of versions greater than 1.0 and us not wanting to have more than one version of go-state-types in the dependency tree.)  

Additional notes:
- We strive for round minor versions (e.g., 0.x.0) to denote the definitive release for a given network upgrade. However, due to the unpredictability of software engineering, further releases may be made by bumping the patch number (e.g., 0.x.1).
- Development versions use qualifiers like `-dev` (development) and `-rc` (release candidate).
```

I'm espousing/guessing why we always set the major version to 0.  Feel free to fix/correct.  I think it's worth giving the rationale though.  (I know I have seen writeup on this in the past in PL but I couldn't find anything to link to in my quick searching.)

Reply:
Updated with some slight changes in the `We don´t use major versions;` bullet point here: https://github.com/filecoin-project/go-state-types/pull/273/commits/cc68b5b1607bcb72d3d9b44fc63cf3ca3c7abd82

Reply:
Reasoning for not bumping major versions in Go is afaict bad because it requires changing import paths for every single package, including internal ones, to include the new version number, e.g., import github.com/filecoin-project/lotus/v2/*.

Some additional notes has been written about this during an earlier disssion for this in the Lotus repo here:
- https://github.com/filecoin-project/lotus/discussions/5889#discussioncomment-601439
- https://github.com/filecoin-project/lotus/discussions/5889#discussioncomment-591545